src/storage/store/string_column.cpp

```diff
@@ -154,54 +157,83 @@ void StringColumn::scanUnfiltered(transaction::Transaction* transaction,
     scanValuesToVector(transaction, nodeGroupIdx, offsetsToScan, resultVector, indexState);
 }
 
+std::ranges::subrange<std::vector<std::pair<StringColumn::string_index_t, uint64_t>>::iterator>
+getNextRange(std::vector<std::pair<StringColumn::string_index_t, uint64_t>>& data,
```

Comment:
Yes, me too. Maybe it could just be replaced by a pair, but it's a little more convenient and helps simplify the code.
https://en.cppreference.com/w/cpp/compiler_support/20 seems to indicate that it's supported by the four major stdlibs (under "The One Ranges Proposal"). [std::ranges::minmax_result](https://en.cppreference.com/w/cpp/algorithm/ranges/minmax) turned out to not be well supported by apple clang (which is why `std::minmax_result was used instead in the old code here). I'm not sure how much of ranges is.